source/agora/node/FullNode.d

```diff
@@ -100,17 +98,25 @@ public class Node : API
     /// Blockstorage
     protected IBlockStorage storage;
 
-    /// Nominator instance
-    protected Nominator nominator;
-
     /// Enrollment manager
     protected EnrollmentManager enroll_man;
 
     /// If a custom genesis block is set it will be stored here
     private immutable Block genesis_block;
 
-    /// Ctor
-    public this (const Config config)
+    /***************************************************************************
+
+        Constructor
+
+        Params:
+            config = Config instance
+            required_peer_keys = if not empty, the set of public keys for which
+                to find the associated nodes in the network and to connect to
+
+    ***************************************************************************/
+
+    public this (const Config config,
+        Set!PublicKey required_peer_keys = Set!PublicKey.init)
```

Comment:
This is something I'd like to move out of the `FullNode` constructor, but I couldn't find a good solution yet.